### PR TITLE
product: product search with search model

### PR DESCRIPTION
**Pull request by xbakhromjon**

### CODE REVIEW

File: cmd/app/root.go [hunk 1]
```diff
@@ -2,16 +2,17 @@ package app
...
+	"golang-project-template/internal/common"
+	"os"
 )
 
 var rootCmd = &cobra.Command{
 	Use: "grpc-server",
 	Run: func(cmd *cobra.Command, args []string) {
 		// Application entrypoint...
 
+		common.SetUpEnv()
```
Comment (khudayberdiyev25) [inline cmd/app/root.go:15]: I guess we do not need such method. check How Samandar importing envs.

File: internal/shop/adapters/product_postgres_repo.go [hunk 2]
```diff
@@ -43,3 +50,118 @@ func (p *productPostgresRepo) FindAllByShopId(shopId int) ([]*domain.Product, er
...
+		err := rows.Scan(&product.Id, &product.Name, &product.Price, &product.ShopId)
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, &product)
+	}
+	return result, nil
+}
+
+func (p *productPostgresRepo) FindAllWithPageable(searchModel domain.ProductSearchModel, pageable common.PageableRequest) (*common.PageableResult[domain.Product], error) {
```
Comment (khudayberdiyev25) [inline internal/shop/adapters/product_postgres_repo.go:88]: well. I think converting result to pageable to happen in usecase layer. here you can use FindAll by adding +1 return param like totalcount int64.

**Review by khudayberdiyev25 — CHANGES_REQUESTED** [review]

File: internal/shop/adapters/product_postgres_repo.go (hunk 2 above)
Comment (xbakhromjon) [reply internal/shop/adapters/product_postgres_repo.go:88]: Yes, you are right

File: cmd/app/root.go (hunk 1 above)
Comment (xbakhromjon) [reply cmd/app/root.go:15]: ok, I will see it.

**Pull request merged**